src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -1178,8 +1180,20 @@ private void ReportObsoleteAndFeatureAvailabilityDiagnostics(MethodSymbol operat
             }
         }
 
-        private bool IsDefaultLiteralAllowedInBinaryOperator(BinaryOperatorKind kind, BoundExpression left, BoundExpression right)
+        private bool IsDefaultLiteralOrNewAllowedInBinaryOperator(BinaryOperatorKind kind, BoundExpression left, BoundExpression right)
         {
+            // The default literal is only allowed with equality operators
+            // and both operands cannot be typeless at the same time.
+            if (left.IsTypelessNew())
+            {
+                return !right.IsLiteralDefaultOrTypelessNew();
```

Comment:
>IsLiteralDefaultOrTypelessNew [](start = 30, length = 29)

what if the `right` is some other typeless epxression (`null`, a lambda, or a typeless tuple like `(null, null)`)?

Reply:
I suspect they're handled elsewhere since we didn't have to deal with those for `default`.
I'll add tests to not regress in this area.